mirrord/layer/src/socket/ops.rs

```diff
@@ -1594,26 +1594,56 @@ pub(super) fn getifaddrs() -> HookResult<*mut libc::ifaddrs> {
         Err(io::Error::from_raw_os_error(result))?;
     }
 
-    // Head of the ipv6 interface addresses list.
-    // Should be freed with `libc::freeifaddrs` before returning.
-    let mut ipv6_head: *mut libc::ifaddrs = std::ptr::null_mut();
+    // Count entries for new_list_start malloc
+    let mut entry_count = 0;
+    let mut count_head: *mut libc::ifaddrs = original_head;
+    unsafe {
+        while let Some(ifaddr) = count_head.as_mut() {
+            entry_count += 1;
+            count_head = ifaddr.ifa_next;
+        }
+    }
```

Comment:
If we take control of the allocations, by using `HashSet` or whatever is appropriate, we can get rid of this and build the C linked list at the end, when we have all the pointers ready.

Reply:
i did suggest that approach, @aviramha didnt think it was necessary

Reply:
Hmm I like it better due to how it would help us avoid pointer math (right?). And we already have this pattern elsewhere, so would be more familiar.

But! Don't let this be a blocker, I'll leave the decision to you.